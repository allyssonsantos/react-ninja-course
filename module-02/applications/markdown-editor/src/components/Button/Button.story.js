'use strict';

import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import Button from './index';

const stories = storiesOf('Button', module);

stories.addDecorator((story) => (
  <div style={{ display: 'flex', height: 40 }}>
    {story()}
  </div>
));

stories.add('Button success', () => (
  <Button onClick={action('success')} kind='success'>
    Success
  </Button>
));

stories.add('Button danger', () => (
  <Button onClick={action('danger')} kind='danger'>
    Danger
  </Button>
));
