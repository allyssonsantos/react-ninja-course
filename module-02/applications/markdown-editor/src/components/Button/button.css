.button {
  border: 0;
  color: #fff;
  cursor: pointer;
  font-weight: 700;
  padding: 0 20px;
  text-transform: uppercase;
  transition: all 0.25s ease-in-out;
}

.button.-danger {
  background-color: red;
}

.button.-danger:hover {
  background-color: darkred;
}

.button.-success {
  background-color: green;
}

.button.-success:hover {
  background: darkgreen;
}
