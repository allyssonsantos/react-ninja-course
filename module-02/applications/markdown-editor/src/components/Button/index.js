'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import css from 'strclass';

import './button.css';

const Button = ({ onClick, children, kind }) => (
  <button onClick={onClick} className={css({ [`-${kind}`]: kind }, 'button')}>
    {children}
  </button>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  kind: PropTypes.oneOf(['success', 'danger'])
};

export default Button;
