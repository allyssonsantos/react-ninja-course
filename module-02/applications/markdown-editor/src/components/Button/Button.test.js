'use strict';

import React from 'react';
import Button from './index';
import renderer from 'react-test-renderer';

it('Should Button default match snapshot', () => {
  const tree = renderer.create(<Button onClick={() => null}>Hello Button</Button>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Should Button success match snapshot', () => {
  const tree = renderer.create(<Button onClick={() => null} kind="success">Hello Button</Button>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Should Button danger match snapshot', () => {
  const tree = renderer.create(<Button onClick={() => null} kind="danger">Hello Button</Button>).toJSON();

  expect(tree).toMatchSnapshot();
});
