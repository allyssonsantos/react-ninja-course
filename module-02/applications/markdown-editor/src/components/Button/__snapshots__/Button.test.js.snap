// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Button danger match snapshot 1`] = `
<button
  className="button -danger"
  onClick={[Function]}
>
  Hello Button
</button>
`;

exports[`Should Button default match snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Hello Button
</button>
`;

exports[`Should Button success match snapshot 1`] = `
<button
  className="button -success"
  onClick={[Function]}
>
  Hello Button
</button>
`;
